src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -5233,6 +5234,74 @@ void JIT_Patchpoint(int* counter, int ilOffset)
 
 #endif // FEATURE_ON_STACK_REPLACEMENT
 
+HCIMPL2(void, JIT_ClassProfile, Object *obj, void* tableAddress)
```

Comment:
Add an method header comment staying that this is a profiling/instrumentation method and needs to be optimized  to minimize the cost of runtime profiling.

We might want to explicitly tell the compiler to optimize this method (in retail builds) as we might not normally gather any C++ PGO data for this case if this feature it isn't enabled by default.